### DISCUSSION TITLE
npx taiko failed

### DISCUSSION THREAD

**Discussion by zaalgol** [getgauge/taiko#1995] [General]
I have a Ubuntu 18.04.2  machine. npm version 6.14.12

successfully instaled node:
`apt update  && apt install ca-certificates` 
`apt install -y curl sudo`
`curl -sL https://deb.nodesource.com/setup_12.x | sudo bash -`
`apt install nodejs -y `

but `npx taiko` failed with output:

    internal/modules/cjs/loader.js:818
      throw err;
      ^
    
    Error: Cannot find module '/root/.npm/_npx/4108/lib/node_modules/taiko/lib/install.js'
        at Function.Module._resolveFilename (internal/modules/cjs/loader.js:815:15)
        at Function.Module._load (internal/modules/cjs/loader.js:667:27)
        at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)
        at internal/main/run_main_module.js:17:47 {
      code: 'MODULE_NOT_FOUND',
      requireStack: []
    }
    npm ERR! code ELIFECYCLE
    npm ERR! errno 1
    npm ERR! taiko@1.2.3 install: `node lib/install.js`
    npm ERR! Exit status 1
    npm ERR!
    npm ERR! Failed at the taiko@1.2.3 install script.
    npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
    
    npm ERR! A complete log of this run can be found in:
    npm ERR!     /root/.npm/_logs/2021-04-19T11_26_46_329Z-debug.log
    Install for [ 'taiko@latest' ] failed with code 1
    
  This is the relevant part from the log file:

    20618 silly install taiko@1.2.3
    20619 info lifecycle taiko@1.2.3~install: taiko@1.2.3
    20620 verbose lifecycle taiko@1.2.3~install: unsafe-perm in lifecycle false
    20621 verbose lifecycle taiko@1.2.3~install: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/root/.npm/_npx/4108/lib/node_modules/taiko/node_modules/.bin:/root/.npm/_npx/4108/lib/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
    20622 verbose lifecycle taiko@1.2.3~install: CWD: /root/.npm/_npx/4108/lib/node_modules/taiko
    20623 silly lifecycle taiko@1.2.3~install: Args: [ '-c', 'node lib/install.js' ]
    20624 silly lifecycle taiko@1.2.3~install: Returned: code: 1  signal: null
    20625 info lifecycle taiko@1.2.3~install: Failed to exec install script
    20626 timing action:install Completed in 306ms
    20627 verbose unlock done using /root/.npm/_locks/staging-69fa5b9511ef97fc.lock for /root/.npm/_npx/4108/lib/node_modules/.staging
    20628 timing stage:rollbackFailedOptional Completed in 925ms
    20629 timing stage:runTopLevelLifecycles Completed in 27642ms
    20630 warn optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.1 (node_modules/taiko/node_modules/chokidar/node_modules/fsevents):
    20631 warn notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
    20632 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Valid OS:    darwin
    20632 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Valid Arch:  any
    20632 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Actual OS:   linux
    20632 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Actual Arch: x64
    20633 warn @babel/plugin-bugfix-v8-spread-parameters-in-optional-chaining@7.13.12 requires a peer of @babel/core@^7.13.0 but none is installed. You must install peer dependencies yourself.
    20634 verbose stack Error: taiko@1.2.3 install: `node lib/install.js`
    20634 verbose stack Exit status 1
    20634 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
    20634 verbose stack     at EventEmitter.emit (events.js:314:20)
    20634 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
    20634 verbose stack     at ChildProcess.emit (events.js:314:20)
    20634 verbose stack     at maybeClose (internal/child_process.js:1022:16)
    20634 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
    20635 verbose pkgid taiko@1.2.3
    20636 verbose cwd /
    20637 verbose Linux 5.4.0-1040-gcp
    20638 verbose argv "/usr/bin/node" "/usr/lib/node_modules/npm/bin/npm-cli.js" "install" "taiko@latest" "--global" "--prefix" "/root/.npm/_npx/4108" "--loglevel" "error" "--json"
    20639 verbose node v12.22.1
    20640 verbose npm  v6.14.12
    20641 error code ELIFECYCLE
    20642 error errno 1
    20643 error taiko@1.2.3 install: `node lib/install.js`
    20643 error Exit status 1
    20644 error Failed at the taiko@1.2.3 install script.
    20644 error This is probably not a problem with npm. There is likely additional logging output above.
    20645 verbose exit [ 1, true ]

Thanks

**Comment by zaalgol** [comment]
I found the solution:

`npm install -g taiko --unsafe-perm --allow-root`

Just need to change openBrowser to
```
await openBrowser ({
         args: [
             '--no-sandbox'
         ]
     });

```

**Reply by zabil** [reply]
Yes, there are a few more options you might want to use which are specified here https://docs.taiko.dev/frequently_asked_questions/#how-do-i-launch-chrome-in-sandbox-mode-on-headless-environments%3F

The original command `npx taiko` might not work on ubuntu unless it's installed as a non root user. You can set a custom location for the npm library 

https://github.com/getgauge/template-js/blob/cb0797fbb38cc4d9f92f5127167892914478ac88/Dockerfile#L21-L22
